packages/server/test/integration/http_requests_spec.coffee

```diff
@@ -1007,6 +1009,44 @@ describe "Routes", ->
           expect(res.body).not.to.include("document.domain = 'github.com'")
           expect(res.body).to.include("</html>")
 
+      ## https://github.com/cypress-io/cypress/issues/1746
+      it "can ungzip utf-8 javascript and inject without corrupting it", ->
+        js = ""
+
+        app = express()
+
+        app.use compression({ chunkSize: 64, threshold: 1 })
+
+        app.get "/", (req, res) =>
+          res.setHeader('content-type', 'application/javascript; charset=UTF-8')
+          res.setHeader('transfer-encoding', 'chunked')
+
+          write = (chunk) =>
+            js += chunk
+            res.write(chunk)
+
+          write("function ")
+          _.times 100, =>
+            write("😡😈".repeat(10))
+          write(" () { }")
+          res.end()
+
+        server = http.createServer(app)
+
+        Promise.fromCallback (cb) =>
+          server.listen(12345, cb)
+        .then =>
+          @rp({
+            url: "http://localhost:12345"
+            gzip: true
+          })
+          .then (res) ->
+            expect(res.statusCode).to.eq(200)
+            expect(res.body).to.deep.eq(js)
+
+            Promise.fromCallback (cb) =>
```

Comment:
This code will NOT execute here if the expectations fail above it - which wouldn't close the server, and could possibly cause additional failures in downstream tests.

If you need to guarantee the server is closed regardless of whether or not the test passes either add a `.finally(...)` or move the server instantiation to a property on the test context like `@server = ...` and close it in an `afterEach` via `if (@server) ...`

Reply:
good point, switched to using a .finally